test/e2e/samples.go

```diff
@@ -80,3 +86,61 @@ func createBuild(namespace string, identifier string, filePath string) {
 
 	Logf("Build %s created", identifier)
 }
+
+// retrieveBuildAndBuildRun will retrieve the build and buildRun
+func retrieveBuildAndBuildRun(namespace string, buildRunName string) (*operator.BuildRun, *operator.Build, error) {
+	f := framework.Global
+	buildRun := &operator.BuildRun{}
+	build := &operator.Build{}
+	err := f.Client.Client.Get(goctx.TODO(), types.NamespacedName{Name: buildRunName, Namespace: namespace}, buildRun)
+	if err != nil {
+		Logf("Failed to get BuildRun %s: %s", buildRunName, err)
+		return nil, nil, err
+	}
+	buildName := buildRun.Spec.BuildRef.Name
+	err = f.Client.Client.Get(goctx.TODO(), types.NamespacedName{Name: buildName, Namespace: namespace}, build)
+	if err != nil {
+		Logf("Failed to get Build %s: %s", buildName, err)
+		return nil, nil, err
+	}
+	return buildRun, build, nil
+}
+
+// outputBuildAndBuildRunStatusAndPodLogs will output the status of build and buildRun, print logs of taskRun pod
+func outputBuildAndBuildRunStatusAndPodLogs(namespace string, buildRunName string,) {
+	f := framework.Global
+	buildRun, build, _ := retrieveBuildAndBuildRun(namespace, buildRunName)
+	Logf("The status of Build %s: %s", build.Name, build.Status.Reason)
+	Logf("The status of BuildRun %s: %s", buildRun.Name, buildRun.Status.Succeeded)
+	Logf("The reason of BuildRun %s: %s", buildRun.Name, buildRun.Status.Reason)
```

Comment:
This does not work. If you ignore the `error` returned by `retrieveBuildAndBuildRun`, then you have nil pointer access in lines 113 to 115.

Reply:
Oh, yes, sorry, I forget it, thanks!